.devcontainer.json

```diff
@@ -0,0 +1,39 @@
+{
+  "name": "terraform-provider-argocd",
+  // officiall MS template from https://github.com/devcontainers/templates/tree/main/src/go
+  "image": "mcr.microsoft.com/devcontainers/go:1.20-bookworm",
```

Comment:
official image from Microsoft containing the specified go environment including `golangci-lint` and VS Code extensions ready to code

Could be replaced with a customized image by referencing a `Dockerfile` instead.